libs/go/libp2p_node/aea/api.go

```diff
@@ -397,23 +425,42 @@ func (aea *AeaApi) stop() {
   Pipes helpers
 
 */
+const CurrentVersion = "0.1.0"
 
-func write_envelope(pipe Pipe, envelope *Envelope) error {
-	data, err := proto.Marshal(envelope)
+func MakeACNMessageFromEnvelope(envelope *Envelope) (error, []byte) {
+	envelope_bytes, err := proto.Marshal(envelope)
+	if err != nil {
+		return err, envelope_bytes
+	}
+	return acn.EncodeACNEnvelope(envelope_bytes)
+}
+
+func (aea AeaApi) SendEnvelope(envelope *Envelope) error {
+	err, data := MakeACNMessageFromEnvelope(envelope)
 	if err != nil {
 		logger.Error().Str("err", err.Error()).Msgf("while serializing envelope: %s", envelope)
 		return err
 	}
-	return pipe.Write(data)
-}
+	err = aea.pipe.Write(data)
+	if err != nil {
+		logger.Error().Str("err", err.Error()).Msgf("on pipe write")
+		return err
+	}
+
+	status, err := acn.WaitForStatus(aea.acn_status_chan, ACN_STATUS_TIMEOUT)
 
-func read_envelope(pipe Pipe) (*Envelope, error) {
-	envelope := &Envelope{}
-	data, err := pipe.Read()
 	if err != nil {
-		logger.Error().Str("err", err.Error()).Msg("while receiving data")
-		return envelope, err
+		logger.Error().Str("err", err.Error()).Msgf("on status wait")
+		return err
 	}
-	err = proto.Unmarshal(data, envelope)
-	return envelope, err
+	if status.Code != acn.Status_SUCCESS {
+		logger.Error().Str("err", err.Error()).Msgf("bad status: %d", status.Code)
```

Comment:
`Method call 'err.Error()' might lead to a nil pointer dereference `

Reply:
done